c7n/resources/appelb.py

```diff
@@ -290,35 +307,35 @@ def process(self, resources, event=None):
 
         for w in wafs:
             if 'c7n:AssociatedResources' not in w:
-                arns = client.list_resources_for_web_acl(
-                    WebACLArn=w['ARN']).get('ResourceArns', [])
+                arns = self.retry(
+                    client.list_resources_for_web_acl,
+                    WebACLArn=w.get('ARN', ''),
+                    ResourceType='APPLICATION_LOAD_BALANCER').get('ResourceArns', [])
                 w['c7n:AssociatedResources'] = arns
             name_arn_map[w['Name']] = w['ARN']
             for r in w['c7n:AssociatedResources']:
                 resource_map[r] = w['ARN']
 
-        target_acl_id = name_arn_map.get(target_acl, target_acl)
+        target_acl_ids = [v for k, v in name_arn_map.items() if
+                          re.match(target_acl, k)]
+
         arn_key = self.manager.resource_type.id
         state_map = {}
         for r in resources:
             arn = r[arn_key]
-            if arn in resource_map:
-                # NLB & GLB doesn't support WAF. So, skip such resources
-                if r['Type'] == 'application':
-                    r['c7n_webacl'] = resource_map[arn]
-                    if not target_acl:
-                        state_map[arn] = True
-                        continue
-                    r_acl = resource_map[arn]
-                    if r_acl == target_acl_id:
-                        state_map[arn] = True
-                        continue
+            # NLB & GLB doesn't support WAF. So, skip such resources
+            if r['Type'] == 'application':
```

Comment:
how about != 'application' continue, and the rest of this block can be dedented.

Reply:
Updated if condition